kie-plugins-testing/src/test/java/org/kie/maven/plugin/KieMavenPluginBaseIntegrationTest.java

```diff
@@ -54,20 +59,39 @@ protected void restoreKiePom(String projectName) throws Exception {
     }
 
     protected MavenExecutionResult buildKJarProject(String kjarName, String[] options, String... mavenGoals) throws Exception {
+        logger.info("buildKJarProject {} {} {}", kjarName, options, mavenGoals);
         final MavenRuntime mavenRuntime = mavenRuntimeBuilder
                 .forkedBuilder()
                 // To enable logging using slf4j-simple on the internal classes of the plug-in:
                 //.withCliOptions("-Dorg.slf4j.simpleLogger.defaultLogLevel=debug")
                 .withEnvironment(System.getenv())
                 .withCliOptions(options)
                 .build();
+        StringBuilder builder = new StringBuilder();
+        builder.append(kjarName).append("_");
+        builder.append(mavenRuntime.getMavenVersion()).append("_");
+        for (String option : options) {
+            builder.append(option).append("_");
+        }
+        for (String mavenGoal : mavenGoals) {
+            builder.append(mavenGoal).append("_");
+        }
+        final String key = builder.toString();
+        if (!BUILT_MAP.containsKey(key)) {
+            BUILT_MAP.put(key, getMavenExecutionResult(kjarName, mavenRuntime, mavenGoals));
+        }
+        return BUILT_MAP.get(key);
```

Comment:
Is it possible to replace these lines with a `computeIfAbsent`?
Using `ConcurrentHashMap` the whole method is [performed atomically](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-) to be thread safe

Reply:
@danielezonca 
Nope because of the exception -: computeIfAbsent was my first attempt